golang/pkg/crane/k8s/secret.go

```diff
@@ -166,6 +171,66 @@ func ApplyRegistryAuthSecret(ctx context.Context,
 	return nil
 }
 
+func (s *Secret) GetValidSecret() (string, error) {
+	namespace := s.appConfig.Namespace
+	name := s.appConfig.SecretName
+
+	if namespace == "" || name == "" {
+		return "", fmt.Errorf("secret name or namespace can't be empty")
```

Comment:
What if one of them is empty?

Reply:
These variables have default values, it can be emptied by the user (with envvars), this check blocks that.